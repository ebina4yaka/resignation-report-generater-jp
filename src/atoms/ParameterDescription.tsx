import { ReactElement } from 'react'
import { createStyles, makeStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'

type Props = {
  name: string
  description: string[]
}

const useStyles = makeStyles((theme) =>
  createStyles({
    root: {
      margin: theme.spacing(2, 0, 2),
    },
  })
)

export default function ParameterDescription(props: Props): ReactElement {
  const { name, description } = props
  const classes = useStyles()
  return (
    <div className={classes.root}>
      <Typography>{name}</Typography>
      {description.map((item) => (
        <div key={item}>{item}</div>
      ))}
    </div>
  )
}
