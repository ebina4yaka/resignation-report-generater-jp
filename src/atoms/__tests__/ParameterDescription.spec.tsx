import renderer from 'react-test-renderer'
import ParameterDescription from '../ParameterDescription'

test('Snapshot test for ParameterDescription', () => {
  const tree = renderer
    .create(
      <ParameterDescription name="test" description={['test1', 'test2']} />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
