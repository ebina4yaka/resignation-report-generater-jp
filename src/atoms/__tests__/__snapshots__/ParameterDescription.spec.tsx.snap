// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for ParameterDescription 1`] = `
<div
  className="makeStyles-root-1"
>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    test
  </p>
  <div>
    test1
  </div>
  <div>
    test2
  </div>
</div>
`;
